File: javaSource/ActivitySample/app/src/main/java/com/guagua/activitysample/MyActivity1.java
package com.guagua.activitysample;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Button;
import android.widget.Toast;

public class MyActivity1 extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_my1);

        Button button = findViewById(R.id.btn1);
        button.setOnClickListener(view -> {
            Toast.makeText(this, "close activity", Toast.LENGTH_SHORT).show();
            finish();
        });

        Button start = findViewById(R.id.startBtn);
        start.setOnClickListener(view -> {
            Intent intent = new Intent(this, MyActivity2.class);
            startActivity(intent);
        });

        Button send = findViewById(R.id.sendBtn);
        send.setOnClickListener(view -> {
            Intent intent = new Intent(this, MyActivity2.class);
            intent.putExtra("nick", "guagua");
            intent.putExtra("age", 19);
            startActivity(intent);
        });

        Button req = findViewById(R.id.requestBtn);
        req.setOnClickListener(view -> {
            Intent intent = new Intent(this, MyActivity2.class);
            startActivityForResult(intent, 1010);
        });

        Button life = findViewById(R.id.lifeBtn);
        life.setOnClickListener(view -> {
            Intent intent = new Intent(this, LifeActivity.class);
            startActivity(intent);
        });

        Button standardBtn = findViewById(R.id.standardBtn);
        standardBtn.setOnClickListener(view -> {
            Intent intent = new Intent(this, StandardActivity.class);
            startActivity(intent);
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (RESULT_OK != resultCode) return;
        if (1010 == requestCode && null != data) {
            String value = data.getStringExtra("data");
            Log.i("MyActivity1", "onActivityResult(" + requestCode + "), result value: " + value);
        }
    }
}

File: kotlinSource/ActivitySample/app/src/main/java/com/guagua/activity_sample/MyActivity1.kt
package com.guagua.activity_sample

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.Toast

class MyActivity1 : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_my1)

        val button: Button = findViewById(R.id.btn1)
        button.setOnClickListener{
            Toast.makeText(this, "close activity", Toast.LENGTH_SHORT).show()
            finish()
        }

        val start: Button = findViewById(R.id.startBtn)
        start.setOnClickListener{
            val intent = Intent(this, MyActivity2::class.java)
            startActivity(intent)
        }

        val send: Button = findViewById(R.id.sendBtn)
        send.setOnClickListener{
            val intent = Intent(this, MyActivity2::class.java)
            intent.putExtra("nick", "guagua")
            intent.putExtra("age", 19)
            startActivity(intent)
        }

        val req: Button = findViewById(R.id.requestBtn)
        req.setOnClickListener{
            val intent = Intent(this, MyActivity2::class.java)
            startActivityForResult(intent, 1010)
        }

        val life: Button = findViewById(R.id.lifeBtn)
        life.setOnClickListener{
            val intent = Intent(this, LifeActivity::class.java)
            startActivity(intent)
        }

        val standardBtn: Button = findViewById(R.id.standardBtn)
        standardBtn.setOnClickListener{
            val intent = Intent(this, StandardActivity::class.java)
            startActivity(intent)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (RESULT_OK != resultCode) return
        if (1010 == requestCode) {
            val value = data?.getStringExtra("data")
            Log.i("MyActivity1", "onActivityResult($requestCode), result value: $value")
        }
    }
}